packages/web-components/src/utils/element-internals.ts

```diff
@@ -33,3 +33,36 @@ export function toggleState(elementInternals: ElementInternals, state: string, f
 
   elementInternals.states.delete(state);
 }
+
+/**
+ * A decorator for toggling the element internals of an element
+ *
+ * @internal
+ */
+export function toggleAttrState(target: any, attr: string): any {
+  // Stash the original method
+  const method = target[`${attr}Changed`];
+
+  // Replace the attrChanged method with a new one
+  target[`${attr}Changed`] = function (prev: string | boolean | undefined, next: string | boolean | undefined): void {
```

Comment:
Maybe both `next` and `prev`’s types can be `any` since you can add custom converters.